scene/gui/control.cpp

```diff
@@ -1845,19 +1845,46 @@ Control::FocusMode Control::get_focus_mode() const {
 	return data.focus_mode;
 }
 
+void Control::set_focus_layer(int p_layer_id) {
+	// If this Control has focus, then lets find the next valid Control to focus before switching layers.
+	if (has_focus()) {
+
+		Control *next = find_next_valid_focus();
+
+		if (next) {
```

Comment:
```suggestion
		if (next && next != this) {
```
Otherwise the control doesn't lose focus properly.